### PR TITLE
Replace MQTT files in C-SDK with submodule to coreMQTT

### PR DESCRIPTION
**Pull request by aggarw13**
Use submodule pointer to **coreMQTT** repositriy, the Single Source of Truth (SSoT) for MQTT stack

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: demos/mqtt/mqtt_demo_plaintext/Makefile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 CC = gcc
 
 INCLUDE_DIRS = -I . \
-               -I ../../../libraries/standard/mqtt/include \
+               -I ../../../libraries/standard/coreMQTT/include \
```
Comment (leegeth) [inline demos/mqtt/mqtt_demo_plaintext/Makefile:4]: This is a good find. I don't think we need this file anymore. Do you know of a reason why we would need this file. This is out of the scope of this PR, but we may need to just remove it.

File: demos/shadow/shadow_demo_main/core_mqtt_config.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
```
Comment (leegeth) [inline demos/shadow/shadow_demo_main/core_mqtt_config.h:1]: Why is this file added as a new file? Wasn't there a mqtt_config.h already?

File: demos/mqtt/mqtt_demo_plaintext/Makefile (hunk 1 above)
Comment (aggarw13) [reply demos/mqtt/mqtt_demo_plaintext/Makefile:4]: https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1182
Had created a PR to remove this unused file once I found it in change set 😃

File: demos/shadow/shadow_demo_main/core_mqtt_config.h (hunk 2 above)
Comment (aggarw13) [reply demos/shadow/shadow_demo_main/core_mqtt_config.h:1]: Nope, it wasn't present before in the Shadow demo. It was instead utilizing the mqtt_config.h file from the connection sharing demo due to configuration logic in the demo's CMakeLists.txt which I have removed as part of this PR. 
Removing the include path config logic from connection sharing demo's CMakeLists.txt exposed that the shadow demo lacked the mqtt config file

Comment (leegeth) [reply demos/shadow/shadow_demo_main/core_mqtt_config.h:1]: Ok. Thanks. That was a good find and a fix.

**Review by cobusve — APPROVED** [review]
